### PR TITLE
Accessibility Updated

### PR DESCRIPTION
**Pull request by msrafi**
JS Fiddle added: http://jsfiddle.net/equinoxs/1x8w9c3n/39/

### CODE REVIEW

**Comment by msrafi** [comment]
@kenwheeler 
Hi, please let me know if this works. Comments are welcome.

File: slick/slick.js [hunk 1]
```diff
@@ -526,9 +526,10 @@
 
         _.updateDots();
 
-        if (_.options.accessibility === true) {
-            _.$list.prop('tabIndex', 0);
-        }
+        // This is controlled through _.activeADA method
+        //if (_.options.accessibility === true) {
+        //    _.$list.prop('tabIndex', 0);
+        //}
```
Comment (oliverbenns) [inline slick/slick.js]: If it's not being used, let's get rid of it, not comment it out.

File: slick/slick.js [hunk 2]
```diff
@@ -45,15 +45,15 @@
                 appendDots: $(element),
                 arrows: true,
                 asNavFor: null,
-                prevArrow: '<button type="button" data-role="none" class="slick-prev" aria-label="previous">Previous</button>',
-                nextArrow: '<button type="button" data-role="none" class="slick-next" aria-label="next">Next</button>',
+               prevArrow: '<button type="button" data-role="none" class="slick-prev" aria-label="Previous" tabindex="0" role="button">Previous</button>',
```
Comment (oliverbenns) [inline slick/slick.js]: Slight indentation fix needed here :+1:

**Comment by oliverbenns** [comment]
Nice work dude!

**Comment by msrafi** [comment]
comments removed

**Comment by msrafi** [comment]
Thanks @oliverbenns

**Comment by kenwheeler** [comment]
For some reason I just saw this,I'll review tonight, thanks!

**Comment by msrafi** [comment]
History: #838 #1146

File: slick/slick.js [hunk 3]
```diff
@@ -2473,3 +2552,14 @@
     };
 
 }));
+
+
+$(function(){
```
Comment (kenwheeler) [inline slick/slick.js]: What's this all about?

File: slick/slick.js [hunk 4]
```diff
@@ -1283,19 +1284,21 @@
...
-                }
-            });
-        } else if (event.keyCode === 39 && _.options.accessibility === true) {
-            _.changeSlide({
-                data: {
-                    message: 'next'
-                }
-            });
+         //Dont slide if the cursor is inside the form fields and arrow keys are pressed
+        if(!event.target.tagName.match('TEXTAREA|INPUT|SELECT')) {
```
Comment (jandersonBB) [inline slick/slick.js]: ++ nice

File: slick/slick.js (hunk 3 above)
Comment (msrafi) [inline slick/slick.js]: My Bad. Kept the sample testing initiation. Removed it now.

File: slick/slick.js (hunk 4 above)
Comment (kenwheeler) [inline slick/slick.js]: nice indeed

**Comment by kenwheeler** [comment]
Well done @msrafi

**Comment by msrafi** [comment]
Cool. Thank you

**Comment by Cottser** [comment]
:clap: thanks everyone! So happy to see this in.

**Comment by leggomuhgreggo** [comment]
Hey, the addition of the `tabindex` attribute is triggering the focus state when a slide has `tabindex="0"`. This is causing the default outline from the user agent styles around the slide.

It turns out that its happening on Chrome 43 [[screenshot](http://i.imgur.com/6VpGsWp.png)], and IE11, but not Firefox 39. I think Firefox may have depricated support at some point, becuase in looking into the issue it seems like they [had it at one point](https://nemisj.com/focusable/), but [don't now](http://stackoverflow.com/questions/30883332/tabbing-throgh-keyboard-not-working-in-mozilla-firefox).

I'm not sure what the best practice is in situations like this. Clearly the focus state is useful for accessibility, but it's obtrusive from a design standpoint. 

I think I'm probably going to add  `.slick-slide:focus{outline:0}` in production, and I wonder if adding it to slick-theme.css a fair middle ground?

[Here's an example](http://codepen.io/gwst/pen/Xbqymp) with the style override commented out.

Thanks!

**Comment by msrafi** [comment]
I added this for screen reader. So when it slides the screen reader will actually read the details about that particular slide. Otherwise there is no option for letting the screen reader to give that input. 

I agree on the chrome outline, but I would recommend leaving the slick css very generic so that the CSS can be updated specific to project requirements.

You can read more #792 on this topic.

Thanks

**Comment by leggomuhgreggo** [comment]
@msrafi So having `outline: 0` in the slick-theme.css but not in slick.css makes sense to you too, then?

After skimming #792 I can see there was definitely some thought put into the tab focus sequencing, but I noticed that when there are multiple slides, the focus state is added to the right-most of the slick-active slides. Might be I'm missing something, but does it made more sense to add the focus state to the first (left-most) of the slide-active slides--at least when the RTL option is false?

I was messing around, and I'm pretty sure just adding `.eq(0)` to the slick-active selection on line [2599](https://github.com/kenwheeler/slick/blob/master/slick/slick.js#L2599) does the trick, where, after scroll, the left-most slide is first to be focused, then subsequent visible ones then next nav. 

Might be worth conditioning based on RTL. That functionality seems messy at the moment, though.

Based on reading the spec, I actually tried to order the tabindexes sequentially so that the left-most slide would have `tabindex=0` and the next slide `tabindex=1` and so on, but, at least in Chrome, the tab sequence went from `tabindex=0` right to the next arrow instead of `tabindex=1`. Not sure why it behaves that way, but I guess all of them having `tabindex=0` works fine.

Thanks

**Pull request merged**